src/compiler/crystal/tools/doc/templates.cr

```diff
@@ -1,6 +1,30 @@
 require "ecr/macros"
 
 module Crystal::Doc
+  SVG_DEFS = <<-SVGS
+<svg class="hidden">
+  <symbol id="octicon-link" viewBox="0 0 16 16">
+    <path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path>
+  </symbol>
+</svg>
+SVGS
+
+  private module Anchorable
```

Comment:
Actually, I don't think this module should exist. It's not very useful, you could just transform the anchor string in `.anchor_link` directly.

Reply:
I don't think I'm a fan of doing any transformations within `.anchor_link`.  It should just be responsible for outputting the markup for a given `anchor`.

Reply:
It's just a generic string transformation that is well within the scope of `anchor_link`.

Reply:
And you'd be okay with doing the extra gsub/downcase calls on every `anchor` even those that are already correct (from the `render_header` method)?

Currently this transformation is just scoped to the built in templates.

Reply:
Sure. Alternatively you could explicitly call a transformation method for those cases, where it't not already a slugified value. I don't care, but I'd want to avoid an additional type just for this when there are suitable alternative solutions.